pageserver/src/http/routes.rs

```diff
@@ -353,6 +354,32 @@ async fn status_handler(
     json_response(StatusCode::OK, StatusResponse { id: config.id })
 }
 
+async fn reload_auth_validation_keys_handler(
+    request: Request<Body>,
+    _cancel: CancellationToken,
+) -> Result<Response<Body>, ApiError> {
+    check_permission(&request, None)?;
+    let config = get_config(&request);
+    let state = get_state(&request);
+    let Some(shared_auth) = &state.auth else {
+        return json_response(StatusCode::BAD_REQUEST, ());
+    };
+    // unwrap is ok because check is performed when creating config, so path is set and exists
+    let key_path = config.auth_validation_public_key_path.as_ref().unwrap();
+    info!("Reloading public key(s) for verifying JWT tokens from {key_path:#?}");
+
+    match JwtAuth::from_key_path(key_path) {
+        Ok(new_auth) => {
+            shared_auth.swap(new_auth);
+            json_response(StatusCode::OK, ())
+        }
+        Err(e) => {
+            warn!("Error reloading public keys from {key_path:#?}: {e:}");
```

Comment:
Does the `:#?` have some special meaning with camino paths?

Reply:
I copy pasted it from the `"Loading public key(s) for verifying JWT tokens from {key_path:#?}"`. The `:#?` [indicates](https://doc.rust-lang.org/nightly/std/fmt/index.html#sign0
) "alternate" debug formatting which generally means extra line breaks and indentation. In this instance, it might actually not make a difference at all, as through many layers the path `Debug` impl boils down to:

https://github.com/rust-lang/rust/blob/698db856de0b67313ddcb96b6599598058489ea9/library/core/src/str/lossy.rs#L78-L108

You don't see the [alternate function](https://doc.rust-lang.org/nightly/std/fmt/struct.Formatter.html#method.alternate) being called to query if we are in alternate mode.

In other words, we can probably remove it.

Reply:
Removed the `:#?` stuff and replaced it with `:?`

Reply:
Yep, the `{:?}` is fine if you want it quoted, which works great here. Before asking I did do some testing but failed to see any difference.

> core/str/lossy.rs

Hmm I thought this was a camino path but got even confused.

Resolving.